### ISSUE TITLE
Generated files are sometimes merged with wrong assemblies

### ISSUE THREAD

**Issue by rezanid** [meziantou/Meziantou.Polyfill#41]
### Version of the Meziantou.Polyfill NuGet package

1.0.39

### Description

I am still trying to figure out what exactly is happening here, but it started when I upgraded from .Net Framework 4.7.2 to 4.8, although I don't think it's the target framework that is causing this issue. It must be something else.

When I build, I get compilation errors like the following:
```
 error CS0518: Predefined type 'System.Runtime.CompilerServices.IsExternalInit' is not defined or imported
```

When I checked the source to see where the files are getting injected to, I found the following.
```csharp
#region Assembly YamlDotNet, Version=16.0.0.0, Culture=neutral, PublicKeyToken=ec19458f3c15af5e
// C:\Users\G99202\.nuget\packages\yamldotnet\16.0.0\lib\net47\YamlDotNet.dll
// Decompiled with ICSharpCode.Decompiler 8.1.1.7464
#endregion

namespace System.Diagnostics.CodeAnalysis;

[AttributeUsage(AttributeTargets.Property | AttributeTargets.Field | AttributeTargets.Parameter, Inherited = false)]
[ExcludeFromCodeCoverage]
[DebuggerNonUserCode]
internal sealed class DisallowNullAttribute : Attribute
{
}
#if false // Decompilation log
'279' items in cache
------------------
Resolve: 'mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089'
Found single assembly: 'mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089'
Load from: 'C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.8\mscorlib.dll'
------------------
Resolve: 'System, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089'
Found single assembly: 'System, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089'
Load from: 'C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.8\System.dll'
------------------
Resolve: 'System.Core, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089'
Found single assembly: 'System.Core, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089'
Load from: 'C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.8\System.Core.dll'
#endif

```

For some reason they types ended up in Yamldotnet and because they are generated as `internal` they aren't accessible to the main assembly being built.

### Reproduction Steps

Not exactly sure, but for me what triggered the issue, was to upgrade from .Net Framework 4.7.2 to 4.8 in a class library. This is the project that it happened to: https://github.com/rezanid/xrmtools/tree/assembly-to-config-file

### Other information

I am using the latest released version of Visual Studio 2022.
Microsoft Visual Studio Professional 2022 (64-bit) - Current
Version 17.11.1

I have the following in my csproj file.
```xml
    <PackageReference Include="Meziantou.Polyfill">
      <Version>1.0.39</Version>
      <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
      <PrivateAssets>all</PrivateAssets>
    </PackageReference>
```

I also noticed I have the same issue with PollySharp

**Comment by rezanid** [comment]
Although the root cause is not clear, my issue has been solved by reverting the change and upgrading to .Net Framwork 4.8 again. The only thing I did differently this time it to open the csproj file and change the `<TargetFrameworkVersion>v4.8</TargetFrameworkVersion>` directly instead of relying on the properties tab of the project. I guess we have to blame this one on VS.

I hope this issue helps someone else out there one day ;)
